src/Inventory/Conf.php

```diff
@@ -822,52 +833,62 @@ public function showConfigForm()
             ]
         );
         echo "</td>";
-        echo "<td>" . _n('Action', 'Actions', 1) . "</td>";
+        echo "<td>" . __('Clean agent') . "</td>";
         echo "<td width='20%'>";
         //action
-        $rand = Dropdown::showFromArray(
-            'stale_agents_action',
-            self::getStaleAgentActions(),
-            [
-                'value' => $config['stale_agents_action'] ?? self::STALE_AGENT_ACTION_CLEAN,
-                'on_change' => 'changestatus();',
-            ]
-        );
-        //if action == action_status => show blocation else hide blocaction
-        echo Html::scriptBlock("
-         function changestatus() {
-            if ($('#dropdown_stale_agents_action$rand').val() != 0) {
-               $('#blocaction1').show();
-               $('#blocaction2').show();
-            } else {
-               $('#blocaction1').hide();
-               $('#blocaction2').hide();
-            }
-         }
-         changestatus();
-
-      ");
+        Dropdown::showYesNo('stale_agents_clean', $config['stale_agents_clean'] ?? 0);
         echo "</td>";
         echo "</tr>";
         //blocaction with status
-        echo "<tr class='tab_bg_1'><td colspan=2></td>";
+        echo "<tr class='tab_bg_1'>";
         echo "<td>";
-        echo "<span id='blocaction1' style='display:none'>";
         echo __('Change the status');
-        echo "</span>";
         echo "</td>";
         echo "<td width='20%'>";
-        echo "<span id='blocaction2' style='display:none'>";
         State::dropdown(
             [
                 'name'   => 'stale_agents_status',
                 'value'  => $config['stale_agents_status'] ?? -1,
-                'entity' => $_SESSION['glpiactive_entity']
+                'entity' => $_SESSION['glpiactive_entity'],
+                'toadd'  => [-1 => __('No change')]
             ]
         );
-        echo "</span>";
         echo "</td>";
-        echo "</tr>";
+
+        $plugin_actions = $PLUGIN_HOOKS[Hooks::STALE_AGENT_CONFIG];
+        $odd = true;
+        $in_row = true;
+        /**
+         * @var string $plugin
+         * @phpstan-var array{label: string, item_action: boolean, render_callback: callable, action_callback: callable}[] $actions
+         */
+        foreach ($plugin_actions as $plugin => $actions) {
+            if (is_array($actions) && \Plugin::isPluginActive($plugin)) {
+                foreach ($actions as $action) {
+                    if (!$odd) {
+                        echo "<tr class='tab_bg_1'>";
+                    }
+                    $field = $action['render_callback']($config);
```

Comment:
I just take a look at https://github.com/pluginsGLPI/uninstall/pull/68 and, even if it is a good idea from UX point of view, it will lead to writing a configuration value in the `inventory` context instead of writing in the expected `plugin::uninstall` context.

There is already a configuration page for this plugin, so I would prefer to move this configuration field in plugin configuration context. The render callback could still be used to display the current configuration (read only mode), with a link to the plugin configuration form, or even a modal form to update the plugin configuration, accessible via a :memo: icon.

@orthagh @trasher, your opinion ?

Reply:
I don't see this hook being used very often beyond this one case, but if having the options in the inventory context isn't desired, then the config field name can be used to also specify the context optionally or try to enforce it.

If the field name is not in the list of defaults and starts with "plugin_uninstall_" for example, then plugin:uninstall is the context and everything to the right is the name.
We can try to enforce this field naming by unsetting the value before it is saved if it contains "plugin" and is not in the list of defaults or just rejecting if it isn't in the defaults regardless of the name.

Having the option be read only in the agent cleanup actions config and linking to the plugin's own page for just a single field seems like bad UX. If the config option has to be shown in the uninstall plugin's own config form, then it should be in addition to the option in the inventory config.

Reply:
Saving the value in the right context could probably be handled in a `Hooks::PRE_ITEM_UPDATE` callback.

Displaying the additionnal field could probably be done using `Hooks::POST_ITEM_FORM` callback. In this case, it could simplify the `Hooks::STALE_AGENT_CONFIG` expected format, which would expect the `action_callback`.

Reply:
That hook doesn't account for different sections within the same form though. It would only look somewhat like it should because the stale agent config happens to be the last section currently.

Reply:
Maybe we can add a starting _  for this kind of field (like we do in other place) and add a method `Config::pre_updateInDB` to remove all fields starting with this char.
The plugin can save (or alter the context) itself the value with PRE_ITEM_UPDATE hook.
Maybe just this behavior (just the plugin intercept the call) is needed that said.

I quickly checked if there any existing data like this in my current DB:
![image](https://user-images.githubusercontent.com/418844/179498314-1773ad7f-58df-4094-bb03-3e25d2288ab2.png)

IMO, none of these are valid one and are in the same scenario where we store unwanted data in the table.

Reply:
> Maybe we can add a starting _ for this kind of field (like we do in other place) and add a method `Config::pre_updateInDB` to remove all fields starting with this char.

This requires some modifications on multiple places, so if we want to do this, I think it could be better to do it in a separated PR, to make the review easier.